Add honor code and identity verification for exams (#2228)
PrairieLearn/PrairieLearn: doc/assessment.md

![Closed assessment with real-time grading disabled](assessment-grading-disabled-closed.png)

Note that after the exam has closed and been graded, more information about points will be visible.

## Honor code

By default, `Exam` assessments require students to certify their identity and pledge an honor code before starting the assessment:

> * I certify that I am `name` and I am allowed to take this assessment in `course`.

Author: ```suggestion
* I certify that I am `name` and I am allowed to take this assessment.
```

> * I certify that I am `name` and I am allowed to take this assessment in `course`.

Reviewer: ```suggestion
* I certify that I am `name` and I am allowed to take this assessment.
```

Author: (had already committed this change before this comment)

* I pledge on my honor that I will not give or receive any unauthorized assistance on this assessment and that all work will be my own.

> To disable this requirement, set `"requireHonorCode": false` as a top-level option in the `infoAssessment.json` file.

Reviewer: ```suggestion
To disable this requirement, set `"requireHonorCode": false` as a top-level option in the `infoAssessment.json` file.

The text of the honor code was based on the University of Maryland's [Honor Pledge](https://www.studentconduct.umd.edu/honor-pledge) and the University of Rochester's [Honor Pledge for Exams](https://www.rochester.edu/college/honesty/instructors/pledge.html). This is a "modified" honor code [McCabe et al., 2002](https://doi.org/10.1023/A:1014893102151), as opposed to "traditional" codes that typically also require students to report any violations of the honor code they observe.
```

![Closed assessment with real-time grading disabled](assessment-grading-disabled-closed.png)

Note that after the exam has closed and been graded, more information about points will be visible.

## Honor code

By default, `Exam` assessments require students to certify their identity and pledge an honor code before starting the assessment:

* I certify that I am `name` and I am allowed to take this assessment.
* I pledge on my honor that I will not give or receive any unauthorized assistance on this assessment and that all work will be my own.

To disable this requirement, set `"requireHonorCode": false` as a top-level option in the `infoAssessment.json` file.

> The text of the honor code was based on the University of Maryland's [Honor Pledge](https://www.studentconduct.umd.edu/honor-pledge) and the University of Rochester's [Honor Pledge for Exams](https://www.rochester.edu/college/honesty/instructors/pledge.html). This is a "modified" honor code [McCabe et al., 2002](https://doi.org/10.1023/A:1014893102151), as opposed to "traditional" codes that typically also require students to report any violations of the honor code they observe.

Reviewer: ```suggestion
The text of the honor code was based on the University of Maryland's [Honor Pledge](https://www.studentconduct.umd.edu/honor-pledge) and the University of Rochester's [Honor Pledge for Exams](https://www.rochester.edu/college/honesty/instructors/pledge.html). This is a "modified" honor code ([McCabe et al., 2002](https://doi.org/10.1023/A:1014893102151)), as opposed to "traditional" codes that typically also require students to report any violations of the honor code they observe.
```

![Closed assessment with real-time grading disabled](assessment-grading-disabled-closed.png)

Note that after the exam has closed and been graded, more information about points will be visible.

## Honor code and identity verification

> By default, `Exam` assessments require the student to agree to an honor code and certify their identity before starting the assessment. To disable this requirement, set `"requireHonorCode": false` as a top-level option in the `infoAssessment.json` file.

Author: ```suggestion
By default, `Exam` assessments require the student to pledge an honor code and certify identity before starting the assessment. To disable this requirement, set `"requireHonorCode": false` as a top-level option in the `infoAssessment.json` file.
```
